### PR TITLE
Add support for operator not

### PR DESCRIPTION
**Pull request by NlightNFotis**

### CODE REVIEW

File: gnat2goto/driver/tree_walk.adb [hunk 1]
```diff
@@ -2791,6 +2794,21 @@ package body Tree_Walk is
...
    end Do_Op_Concat;
 
+   -------------------------
+   --     Do_Not_Op       --
+   -------------------------
+
+   function Do_Not_Op (N : Node_Id) return Irep is
+      boolean_value : constant Irep := Do_Expression (Right_Opnd (N));
+      Op_Kind       : constant Irep_Kind := I_Op_Not;
+      Ret           : constant Irep := New_Irep (Op_Kind);
```
Comment (hannes-steffenhagen-diffblue) [inline gnat2goto/driver/tree_walk.adb]: Sorry for being a bit of a pain on this, but can we use `Make_Op_Not` here? (Generally prefer the `Make_XXX` constructor-type functions over setting all attributes manually)

File: testsuite/gnat2goto/tests/not_op/not_op.adb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+procedure not_op is
+  x : boolean := false;
+begin
+  if not x then
+    x := true;
+    pragma Assert (x = True);
+  else
+    x := false;
```
Comment (hannes-steffenhagen-diffblue) [inline testsuite/gnat2goto/tests/not_op/not_op.adb]: :pick: `pragma Assert (x)` instead?

File: gnat2goto/driver/tree_walk.adb [hunk 3]
```diff
@@ -183,6 +183,9 @@ package body Tree_Walk is
    function Do_Operator_General (N : Node_Id) return Irep
    with Pre  => Nkind (N) in N_Op;
 
+   function Do_Not_Op (N : Node_Id) return Irep
```
Comment (hannes-steffenhagen-diffblue) [inline gnat2goto/driver/tree_walk.adb]: :pick: `Do_Op_Not`, not `Do_Not_Op`

File: gnat2goto/driver/tree_walk.adb (hunk 1 above)
Comment (NlightNFotis) [reply gnat2goto/driver/tree_walk.adb]: Yes, didn't know about it. I'll change it.

**Pull request merged**